metabase/metabase
# Fix: Create api token modal now shows error from backend

For example if name is taken. 

Closes https://github.com/metabase/metabase/issues/47552

### Description

The modal for creating an API token didn't take into consideration whether the request was successful or not. If we got an error from backend, the next step (showing the API token) was shown (independent of if a token was generated or not).

This change now renders the error message coming from backend, if any. Giving the user a chance to change the name and try again. 

Remaning issue: The submit button still says "Success" (with green background) for a few seconds, even if there was an error. 

I also took the opportunity to disable the submit button if the request is in the air (i.e. `response.isLoading`)

### How to verify

1. Go to 'Admin settings -> Authentication'
2. Click "Manage" button in the 'API token' section.
3. Click "Create API Key" and generate a token with name "Foo".
4. Again, click "Create API Key" and generate another token with the same name, "Foo".
5. The error message should now show up next to the Submit button saying: "An API key with this name already exists."  

### Demo

<img width="489" alt="image" src="https://github.com/user-attachments/assets/ff086df4-c5e5-44ca-bf2d-f3be664c7e33">

### Checklist
